Jim-Hodapp-Coaching/refactor-platform-rs
# Create and document an MVP DB schema using DBML

## Description
This PR creates and documents an MVP DB schema for refactor_platform_rs using [DBML](https://dbml.dbdiagram.io/home/).


#### GitHub Issue: Fixes #8 

### Changes
* Creates/documents an MVP DB schema for refactor_platform_rs using DBML
* Adds a script to automatically generate Postgresql-compatible SQL and generate a new DB within a running instance of Postgres
* Updates the `users` entity with new fields
* Uses autoincrement integer-based internal primary keys and external UUID-based ids

DB schema diagram generated using dbdiagram.io's export feature:

![Refactor Coaching Platform Schema v1](https://github.com/Jim-Hodapp-Coaching/refactor-platform-rs/assets/3219120/d8e25a4d-376e-40aa-99de-532f8be06fb0)

### Testing Strategy
1. Drop it in a new diagram session at [dbdiagram.io](https://dbdiagram.io/) ensuring there are no syntactical errors
2. Run `scripts/rebuild_db.sh` and make sure that it creates a new refactor_platform DB in Postgres
3. Ensure you're able to log in using the refactor-platform-fe login page

### Concerns
None
